widget/embedded/src/store/slices/settings.ts

```diff
@@ -177,4 +185,26 @@ export const createSettingsSlice: StateCreator<
       }),
     });
   },
+  setCustomToken: (token) =>
+    set((state) => ({
+      _customTokens: [token, ...state._customTokens],
+    })),
+  deleteCustomToken: (token) =>
+    set((state) => ({
+      _customTokens: state._customTokens.filter(
+        (customToken) => customToken.address !== token.address
+      ),
+    })),
+  customTokens: () => {
+    const customTokens = get()._customTokens;
+    const config = get().config;
+    const supportedBlockchainsFromConfig = getSupportedBlockchainsFromConfig({
```

Comment:
Seems you are repeating `data.ts`'s `blockchains` method here. Couldn't you use that method here? `.blockchains({type: "custom-tokens"})`

Reply:
Yes, I used that method here.